ledger-tool/src/main.rs

```diff
@@ -1065,19 +1065,27 @@ fn load_bank_forks(
         })
     };
 
-    if let Some(halt_slot) = process_options.halt_at_slot {
-        if halt_slot < starting_slot {
-            eprintln!(
+    match process_options.halt_at_slot {
+        // There are no needs to do the following checks for Sentinel value of `Some(0)` and None
+        // of halt_at_slot, because for Sentinel value of `Some(0)`, no transaction will be
+        // replayed, and for `None`, which implies `Some(Slot::MAX), all transactions that are
+        // available will be replayed.
+        None | Some(0) => {}
+        Some(halt_slot) => {
+            if halt_slot < starting_slot {
+                eprintln!(
                 "Unable to load bank forks at slot {halt_slot} because it is less than the starting slot {starting_slot}. \
                 The starting slot will be the latest snapshot slot, or genesis if --no-snapshot flag specified or no snapshots found."
             );
-            exit(1);
-        }
-        // Check if we have the slot data necessary to replay from starting_slot to <= halt_slot.
-        //  - This will not catch the case when loading from genesis without a full slot 0.
-        if !blockstore.slot_range_connected(starting_slot, halt_slot) {
-            eprintln!("Unable to load bank forks at slot {halt_slot} due to disconnected blocks.",);
-            exit(1);
+                exit(1);
+            }
+            // Check if we have the slot data necessary to replay from starting_slot to <= halt_slot.
```

Comment:
```suggestion
            // Check if we have the slot data necessary to replay from starting_slot to >= halt_slot.
```
Just a swap on the sign; we stop once have hit or surpassed halt_slot